src/bp/core/services/middleware/state-manager.ts

```diff
@@ -81,8 +81,24 @@ export class StateManager {
     }
 
     const state = event.state
+    const { result: user, created } = await this.userRepo.getOrCreate(event.channel, event.target)
+    if (created) {
+      process.BOTPRESS_EVENTS.emit('core.analytics', [
+        {
+          botId: event.botId,
+          channel: event.channel,
+          metric: sdk.AnalyticsMetric.NewUsersCount,
+          method: sdk.AnalyticsMethod.IncrementDaily
+        },
+        {
+          botId: event.botId,
+          channel: event.channel,
+          metric: sdk.AnalyticsMetric.TotalUsers,
+          method: sdk.AnalyticsMethod.IncrementTotal
+        }
+      ])
+    }
```

Comment:
The user will count as a new user in the first bot he visits, but he will never be counted as a new user in subsequent ones

Reply:
that's the point so we may know how many new users were created

Reply:
Yeah, but it will not log anything for other bots. Let's say I have "welcome-bot", "flying" and "smalltalk", 

My brand new userId will count as a "one new user created" on welcome-bot. But if I use that same userId to visit flying or smalltalk, it will not log any created user.

That data will be useless, since it will only track "what was the first bot used by the user which caused him to create an account"